app_dart/lib/src/model/firestore/ci_staging.dart

```diff
@@ -0,0 +1,157 @@
+// Copyright 2024 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:cocoon_service/src/service/logging.dart';
+import 'package:github/github.dart';
+
+import '../../service/firestore.dart';
+import 'package:googleapis/firestore/v1.dart' hide Status;
+
+/// Representaion the current work scheduled for a given stage of monorepo check runs.
+///
+/// Staging is required so that engine build artifacts can be made available to any tests that
```

Comment:
Can you call out explicitly what "staging" means here, as elsewhere it usually means our non-prod environment.

Reply:
/// 'Staging' is the breaking up of the CI tasks such that some are performed before others.
/// This is required so that engine build artifacts can be made available to any tests that
/// depend on them.